iis-workflows/iis-workflows-sentimentanalysis/src/main/java/eu/dnetlib/iis/workflows/sentimentanalysis/SentimentAnalyzer.java

```diff
@@ -0,0 +1,192 @@
+import org.apache.avro.mapred.AvroKey;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.apache.log4j.Logger;
+
+import com.google.common.base.Preconditions;
+
+import eu.dnetlib.iis.metadataextraction.schemas.ReferenceBasicMetadata;
+import eu.dnetlib.iis.metadataextraction.schemas.ReferenceMetadata;
+import eu.dnetlib.iis.sentimentanalysis.schemas.DocumentTextWithReferences;
+import eu.dnetlib.iis.sentimentanalysis.schemas.ReferenceWithSentimentLabels;
+import pl.edu.icm.cermine.ComponentFactory;
+import pl.edu.icm.cermine.ContentExtractor;
+import pl.edu.icm.cermine.bibref.model.BibEntry;
+import pl.edu.icm.cermine.bibref.sentiment.model.CiTOProperty;
+import pl.edu.icm.cermine.bibref.sentiment.model.CitationSentiment;
+
+/**
+ * Sentiment analysis module. Facade to cermine's sentiment extraction.
+ * 
+ * @author mhorst
+ */
+public class SentimentAnalyzer extends
+        Mapper<AvroKey<DocumentTextWithReferences>, NullWritable, AvroKey<ReferenceWithSentimentLabels>, NullWritable> {
+
+    private final Logger log = Logger.getLogger(this.getClass());
+
+    @Override
+    protected void map(AvroKey<DocumentTextWithReferences> key, NullWritable value, Context context)
+            throws IOException, InterruptedException {
+        DocumentTextWithReferences input = key.datum();
+        try {
+            ContentExtractor extractor = new ContentExtractor();
+            // currently using random implementation
+            extractor.getConf().setCitationSentimentAnalyser(ComponentFactory.getRandomCitationSentimentAnalyser());
+            extractor.setRawFullText(input.getText().toString());
+            extractor.setReferences(convertReferences(input.getReferences()));
+            List<ReferenceWithSentimentLabels> results = convertCitationSentiment(input.getId(), input.getReferences(),
+                    extractor.getCitationSentiments());
+            for (ReferenceWithSentimentLabels result : results) {
+                context.write(new AvroKey<ReferenceWithSentimentLabels>(result), NullWritable.get());
+            }
+        } catch (Exception e) {
+            log.error("exception occurred when extracting sentiment for document: " + input.getId(), e);
+        }
+    }
+
+    /**
+     * Converts citation sentiments into {@link ReferenceWithSentimentLabels}
+     * objects.
+     * 
+     * @param publicationId
+     * @param references
+     * @param sentiments
+     * @return citation sentiments converted into
+     *         {@link ReferenceWithSentimentLabels} objects
+     */
+    protected static List<ReferenceWithSentimentLabels> convertCitationSentiment(CharSequence publicationId,
+            List<ReferenceMetadata> references, List<CitationSentiment> sentiments) {
+        Preconditions.checkArgument(references.size() != sentiments.size(), "number of sentiments: " + sentiments.size()
+                + " is not equal the number of input references: " + references.size());
```

Comment:
Please use `Preconditions.checkArgument(references.size() != sentiments.size(), "number of sentiments: %s is not equal the number of input references: %s", sentiments.size(), references.size())`, otherwise the string will be constructed every time the method is called. I had a problem like this before and it ended in a huge performance hit.